blockchain-base/src/lib.rs

```diff
@@ -107,6 +108,9 @@ pub trait AbstractBlockchain<'env>: Sized + Send + Sync {
     fn head_hash_from_store(&self, txn: &ReadTransaction) -> Option<Blake2bHash>;
 
     fn get_accounts_chunk(&self, prefix: &str, size: usize, txn_option: Option<&Transaction>) -> Option<AccountsTreeChunk>;
+
+    // TODO: Currently, we can implement request responses in the ConsensusAgent only for *both* protocols, which is why AbstractBlockchain needs to support this.
+    fn get_epoch_transactions<B, F: Fn(&BlockchainTransaction) -> B>(&self, epoch: u32, f: F, txn_option: Option<&Transaction<'env>>) -> Option<Vec<B>>;
```

Comment:
Let's just define that for Nimiq 1.0 epoch n contains only block n. Problem solved. Not that anyone would use this method than for Nimiq 1.0 though :smile: